demo/src/main/java/com/novoda/storagepathfinder/demo/LandingActivity.kt

```diff
@@ -0,0 +1,44 @@
+package com.novoda.storagepathfinder.demo
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import com.novoda.storagepathfinder.StoragePath
+import kotlinx.android.synthetic.main.activity_landing.*
+import java.util.*
+
+class LandingActivity : AppCompatActivity() {
+
+    private lateinit var assetCloner: AssetCloner
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_landing)
+
+        assetCloner = DemoDependenciesFactory.createAssetCloner(applicationContext)
+        val storageInspector = DemoDependenciesFactory.createStorageInspector(applicationContext)
+
+        val deviceStoragePaths = ArrayList<StoragePath>()
+        deviceStoragePaths.add(storageInspector.primaryStorageBasePath)
+        deviceStoragePaths.add(storageInspector.primaryStorageApplicationPath)
+        deviceStoragePaths.addAll(storageInspector.secondaryStorageBasePath)
+        deviceStoragePaths.addAll(storageInspector.secondaryStorageApplicationPath)
```

Comment:
There is no need to create an `ArrayList` like that in kotlin, the standard is to use `listOf`. It uses `ArrayList under the hood: https://github.com/JetBrains/kotlin/blob/1.2.50/libraries/stdlib/src/kotlin/collections/Collections.kt#L76
For this one we can do:
```
val deviceStoragePaths = with(storageInspector) {
  listOf(primaryStorageBasePath, primaryStorageApplicationPath) 
    + secondaryStorageBasePath 
    + secondaryStorageApplicationPath
}
```